### PR TITLE
Remove chrome widgets

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/30653
Resolves https://github.com/brave/brave-browser/issues/30654

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Review by AlexeyBarabash — APPROVED** [review]
@deeppandya could you please fix presubmit warnings and errors
```
Running Python 3 presubmit upload checks ...
** Presubmit Warnings: 2 **
  android/java/org/chromium/chrome/browser/quickactionsearchwidget/BraveQuickActionSearchWidgetProvider.java:14: Public classes and interfaces require JavaDoc comments.
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java:1187: 'else' construct must use '{}'s.
  build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java:10: Public classes and interfaces require JavaDoc comments.
  build/android/bytecode/java/org/brave/bytecode/BraveQuickActionSearchWidgetProviderClassAdapter.java:10: Public classes and interfaces require JavaDoc comments.

Found 6 lines longer than 80 characters (first 5 shown).
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java, line 348, 104 chars
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java, line 350, 140 chars
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java, line 352, 142 chars
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java, line 354, 109 chars
  android/javatests/org/chromium/chrome/browser/BytecodeTest.java, line 517, 102 chars

** Presubmit ERRORS: 1 **
  android/java/org/chromium/chrome/browser/quickactionsearchwidget/BraveQuickActionSearchWidgetProvider.java:12:1: Redundant import from the same package - org.chromium.chrome.browser.quickactionsearchwidget.QuickActionSearchWidgetProvider.
```


All the rest  :+1:

**Comment by deeppandya** [comment]
@AlexeyBarabash I am pushing the commit to resolve it right away.

File: android/java/org/chromium/chrome/browser/quickactionsearchwidget/BraveQuickActionSearchWidgetProvider.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public abstract class BraveQuickActionSearchWidgetProvider extends QuickActionSearchWidgetProvider {
+    public static void setWidgetEnabled(
+            boolean shouldEnableQuickActionSearchWidget, boolean shouldEnableDinoVariant) {
+        // Keeping function body empty as we don't need to enable widgets
+    }
+
+    public static void setWidgetComponentEnabled(
+            @NonNull Class<? extends QuickActionSearchWidgetProvider> component,
+            boolean shouldEnableWidgetComponent) {
+        assert false : "setWidgetComponentEnabled should be redirected to parent in bytecode!";
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/quickactionsearchwidget/BraveQuickActionSearchWidgetProvider.java:23]: I don't see where it is redirected, it seems to be just never used.

File: patches/chrome-android-java-AndroidManifest.xml.patch [hunk 2]
```diff
@@ -74,6 +74,24 @@ index 6f3722cd64029848c1737ced63889cf7cb41c5aa..be7f48b3d844db1eef8419ee31635a53
              <meta-data android:name="android.app.searchable"
                  android:resource="@xml/searchable" />
          </activity-alias>
+@@ -913,7 +928,7 @@ by a child template that "extends" this file.
+         </service>
+ 
+         <!-- Quick Action Search widget -->
+-        <receiver
++        <!-- <receiver
```
Comment (samartnik) [inline patches/chrome-android-java-AndroidManifest.xml.patch:82]: AFAIK in patches it's better to add new string then modify existing.

**Review by samartnik — APPROVED** [review]
lgtm

**Comment by bsclifton** [comment]
Closing as stale. @deeppandya if we still want to do this we can resubmit PR. If not, maybe we can close the issues?

**Pull request closed without merging**